src/main/java/io/cdap/plugin/facebook/source/batch/FacebookBatchSource.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.facebook.source.batch;
+
+import com.facebook.ads.sdk.AdsInsights;
+import com.google.common.base.Preconditions;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.batch.Input;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.dataset.lib.KeyValue;
+import io.cdap.cdap.etl.api.Emitter;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.batch.BatchSource;
+import io.cdap.cdap.etl.api.batch.BatchSourceContext;
+import io.cdap.plugin.common.LineageRecorder;
+import io.cdap.plugin.facebook.source.common.AdsInsightsTransformer;
+import org.apache.hadoop.io.NullWritable;
+
+import java.util.stream.Collectors;
+
+/**
+ * Plugin returns records from Facebook Insights API.
+ */
+@Plugin(type = BatchSource.PLUGIN_TYPE)
+@Name(FacebookBatchSource.NAME)
+@Description("Reads data from Facebook Insights API.")
+public class FacebookBatchSource extends BatchSource<NullWritable, AdsInsights, StructuredRecord> {
+  public static final String NAME = "FacebookAdsBatchSource";
+
+  private final FacebookBatchSourceConfig config;
+
+  public FacebookBatchSource(FacebookBatchSourceConfig config) {
+    this.config = config;
+
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    config.validate();
```

Comment:
you should use new validation API: https://wiki.cask.co/display/CE/Guidelines+on+Plugin+Validation.
For example: https://github.com/data-integrations/dynamodb-source/pull/7